lib/matplotlib/figure.py

```diff
@@ -1528,635 +965,1865 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
+            in_layout=False, transform=self.transSubfigure)
+        self.patch.set_antialiased(False)
+
+        if parent._layoutgrid is not None:
+            self.init_layoutgrid()
+
+    def _redo_transform_rel_fig(self, bbox=None):
+        """
+        Make the transSubfigure bbox relative to Figure transform.
+
+        Parameters
+        ----------
+        bbox : bbox or None
+            If not None, then the bbox is used for relative bounding box.
+            Otherwise it is calculated from the subplotspec.
+        """
+
+        if bbox is not None:
+            self.bbox_relative.p0 = bbox.p0
+            self.bbox_relative.p1 = bbox.p1
+            return
+
+        gs = self._subplotspec.get_gridspec()
+        # need to figure out *where* this subplotspec is.
+        wr = gs.get_width_ratios()
+        hr = gs.get_height_ratios()
+        nrows, ncols = gs.get_geometry()
+        if wr is None:
+            wr = np.ones(ncols)
+        else:
+            wr = np.array(wr)
+        if hr is None:
+            hr = np.ones(nrows)
+        else:
+            hr = np.array(hr)
+        widthf = np.sum(wr[self._subplotspec.colspan]) / np.sum(wr)
+        heightf = np.sum(hr[self._subplotspec.rowspan]) / np.sum(hr)
+
+        x0 = 0
+        if self._subplotspec.colspan[0] > 0:
+            x0 += np.sum(wr[self._subplotspec.colspan[0] - 1]) / np.sum(wr)
+
+        y0 = 0
+        if self._subplotspec.rowspan[-1] < nrows - 1:
+            y0 += 1 - np.sum(hr[self._subplotspec.rowspan[-1]-1]) / np.sum(hr)
+
+        if self.bbox_relative is None:
+            self.bbox_relative = Bbox.from_bounds(x0, y0, widthf, heightf)
+        else:
+            self.bbox_relative.p0 = (x0, y0)
+            self.bbox_relative.p1 = (x0 + widthf, y0 + heightf)
+
+    def get_constrained_layout(self):
+        return self._parent.get_constrained_layout()
+
+    def get_constrained_layout_pads(self, relative=False):
+        return self._parent.get_constrained_layout_pads(relative=relative)
+
+    def init_layoutgrid(self):
+        """Initialize the layoutgrid for use in constrained_layout."""
+        if self._layoutgrid is None:
+            gs = self._subplotspec.get_gridspec()
+            parent = gs._layoutgrid
+            if parent is not None:
+                self._layoutgrid = layoutgrid.LayoutGrid(
+                    parent=parent,
+                    name=(parent.name + '.' + 'panellb' +
+                          layoutgrid.seq_id()),
+                    parent_inner=True,
+                    nrows=1, ncols=1,
+                    parent_pos=(self._subplotspec.rowspan,
+                                self._subplotspec.colspan))
 
-            Parameters
-            ----------
-            gs : GridSpec
+    def get_axes(self):
+        """
+        Return a list of axes in the SubFigure. You can access and modify the
+        axes in the Figure through this list.
```

Comment:
This leaves out the admonition to not modify the list and the equivalency to `.SubFigure.axes`.